src/Buckle/Buckle/CodeAnalysis/Evaluator.cs

```diff
@@ -7,17 +7,44 @@
 
 namespace Buckle.CodeAnalysis;
 
+/// <summary>
+/// Encased object that can also be a reference to a symbol.
+/// </summary>
+internal sealed class EvaluatorObject {
+    internal object value { get; set; }
+    internal bool isReference { get; set; }
+    internal VariableSymbol reference { get; set; }
+
+    internal EvaluatorObject(object value) {
+        this.value = value;
+        this.isReference = false;
+        this.reference = null;
+    }
+
+    internal EvaluatorObject(VariableSymbol reference) {
+        this.value = null;
+
+        if (reference == null) {
+            this.isReference = false;
+            this.reference = null;
+        } else {
+            this.isReference = true;
+            this.reference = reference;
+        }
+    }
+}
+
```

Comment:
This wrapper class is put in place of where Object was used. Instead of always having a value, it can instead also just point to a VariableSymbol (acting as a reference).